### PR TITLE
feat: [SMP-1839]: External Secrets Support for CG, NG License

### PR DESCRIPTION
**Pull request by saitejavn-harness**
## Describe your changes
Added support for external K8S secrets for Harness Licenses: [Doc](https://harness.atlassian.net/wiki/spaces/~7120205e3981db61bd4dcc8068ae7262d845f7/pages/21532803388/external-secrets+External+Secrets+Support+for+Harness+Licenses)

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54451)

### CODE REVIEW

File: charts/bootstrap/secrets/templates/external-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{{- if eq (include "harnesscommon.secrets.hasESOSecrets" (dict "secretsCtx" .Values.global.license.secrets)) "true" }}
```
Comment (kapilgarg1996) [inline charts/bootstrap/secrets/templates/external-secret.yaml:1]: can you share its template ?

Comment (saitejavn-harness) [reply charts/bootstrap/secrets/templates/external-secret.yaml:1]: Added here: [Link](https://harness.atlassian.net/wiki/spaces/~7120205e3981db61bd4dcc8068ae7262d845f7/pages/21532803388/external-secrets+External+Secrets+Support+for+Harness+Licenses#ESO)

File: 120-ng-manager/chart/Chart.lock [hunk 2]
```diff
@@ -1,6 +1,6 @@
 dependencies:
 - name: harness-common
   repository: https://harness.github.io/helm-common
-  version: 1.1.6
-digest: sha256:1fce4dce480c4e0bb5680d222eddaee3f54b2efc4616cdea258c2b47650a87d2
-generated: "2023-10-09T14:12:49.062761+05:30"
+  version: 1.1.9
+digest: sha256:807c35d597e8a1cfa4dbe1245f8ac37babce8b3c2f196b6edccdb44b7f4075fc
```
Comment (arya-harness) [inline 120-ng-manager/chart/Chart.lock:5]: Increment patch version in chart.yaml

File: 360-cg-manager/chart/Chart.lock [hunk 3]
```diff
@@ -1,6 +1,6 @@
 dependencies:
 - name: harness-common
   repository: https://harness.github.io/helm-common
-  version: 1.1.6
-digest: sha256:1fce4dce480c4e0bb5680d222eddaee3f54b2efc4616cdea258c2b47650a87d2
-generated: "2023-10-09T14:15:27.783567+05:30"
+  version: 1.1.9
+digest: sha256:807c35d597e8a1cfa4dbe1245f8ac37babce8b3c2f196b6edccdb44b7f4075fc
```
Comment (arya-harness) [inline 360-cg-manager/chart/Chart.lock:5]: Increment patch version in chart.yaml

**Review by kapilgarg1996 — APPROVED** [review]

**Comment by saitejavn-harness** [comment]
trigger uts

**Comment by saitejavn-harness** [comment]
trigger uts

**Comment by saitejavn-harness** [comment]
trigger uts

**Pull request merged**